Formula/range-v3.rb

```diff
@@ -0,0 +1,34 @@
+class RangeV3 < Formula
+  desc "Experimental range library for C++11/14/17"
+  homepage "https://ericniebler.github.io/range-v3/"
+  url "https://github.com/ericniebler/range-v3/archive/0.3.6.tar.gz"
+  sha256 "ce6e80c6b018ca0e03df8c54a34e1fd04282ac1b068cd39e902e2e5201ac117f"
+
+  depends_on "cmake" => :build
+
+  def install
+    system "cmake", ".",
+                    "-DRANGE_V3_TESTS=OFF",
+                    "-DRANGE_V3_HEADER_CHECKS=OFF",
+                    "-DRANGE_V3_EXAMPLES=OFF",
+                    "-DRANGE_V3_PERF=OFF",
+                    *std_cmake_args
+    system "make", "install"
+  end
+
+  test do
+    (testpath/"test.cpp").write <<~EOS
+      #include <range/v3/all.hpp>
+      #include <iostream>
+      #include <string>
+      using std::cout;
+      int main() {
+        std::string s{ "hello" };
+        ranges::for_each( s, [](char c){ cout << c << " "; });
+        cout << "\n";
+      }
+    EOS
+    system ENV.cc, "-std=c++11", "-o", "test", "test.cpp"
+    assert_equal "./test", 0
```

Comment:
Pretty sure if you just use `system "./test"` it'll let you know sharpish if the exit code isn't `0`.

Reply:
I wish GitHub wouldn't compress old conversational diffs sometimes; apologies for treading on your advice @fxcoudert. We should be asserting on something that `system "./test"` wasn't already doing though, I assume FX was thinking output rather than exit code.

Reply:
The idea behind the test is to check for compilation. I don’t want to check that Range v3 is implemented correctly, only that it is properly installed and can be used to compile a simple program (that program is actually one of the examples taken from the website). I want to see that I can create a valid, runnable binary using the library.

Reply:
Checking that the output is as expected is one more line, something like:
`assert_equal "h e l l o \n", shell_output("./test")`